lib/jake/array.ex

```diff
@@ -19,15 +21,19 @@ defmodule Jake.Array do
       items = Stream.concat([items, additional_items])
 
       StreamData.bind(StreamData.integer(min_items..max_items), fn count ->
-        Enum.take(items, count)
-        |> Enum.map(&Jake.gen(&1))
+        list = Enum.take(items, count)
+
+        for(n <- list, do: %{context | child: n})
+        |> Enum.map(&Jake.gen_lazy(&1))
```

Comment:
do we need to loop twice? can't we just call `gen_lazy(%{context | child: &1})`

Reply:
> can't we just call gen_lazy(%{context | child: &1})

sure